src/dialog/unix/SDL_portaldialog.c

```diff
@@ -234,7 +236,18 @@ static DBusHandlerResult DBus_MessageFilter(DBusConnection *conn, DBusMessage *m
                     goto cleanup;
                 }
             }
-            dbus->message_iter_get_basic(&uri_entry, path + current);
+
+            dbus->message_iter_get_basic(&uri_entry, &uri);
+
+            /* https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.FileChooser.html */
+            /* Returned paths will always start with 'file://'; truncate it */
+            if (SDL_strncmp(uri, "file://", SDL_strlen("file://"))) {
+                path[current] = uri + SDL_strlen("file://");
+            } else if (SDL_strstr(uri, "://")) {
```

Comment:
I don't think we should do this. Even if SDL doesn't support it, the application might.

Reply:
Currently, the file chooser will always return a `file://` protocol, so that check is essentially dead code, but I chose to put it anyways to keep the returned path format consistent should it decide to return other protocols. The API promises to return file paths in OS-specific notation, and some protocols may not be convertible to a local path.

I've chosen to interrupt the execution rather than let it flow silently to avoid funny stuff from happening when developers don't expect a URI, like in #9863, and only then let SDL developers decide what to do with the new protocol. Since the API only promises file paths, and I believe XDG portals are the only desktop implementation that support anything else than a file path, I didn't consider it relevant for an application that uses SDL to expect URIs. If it's preferable to return the URIs even despite the consistency, I can remove the extra check.

Reply:
Fair enough, let's go with this as-is and we can revisit later if needed.